securedrop/journalist_app/sessions.py

```diff
@@ -0,0 +1,283 @@
+        val = self.serializer.dumps(dict(session))
+        if session.to_regenerate:
+            self.redis.delete(self.key_prefix + session.sid)
+            session.sid = self._generate_sid()
+            session.token = self._get_signer(app).dumps(session.sid)  # type: ignore
+        if session.new or session.to_regenerate:
+            self.redis.setex(name=self.key_prefix + session.sid, value=val, time=expires)
+        elif session.modified:
+            # To prevent race conditions where session is delete by an admin in the middle of a req
+            # accept to save the session object if and only if alrady exists using the xx flag
+            self.redis.set(name=self.key_prefix + session.sid, value=val, ex=expires, xx=True)
+        if not session.is_api and (session.new or session.to_regenerate):
+            response.headers.add("Vary", "Cookie")
+            response.set_cookie(
+                app.session_cookie_name,
+                session.token,
+                httponly=httponly,
+                domain=domain,
+                path=path,
+                secure=secure,
+                **conditional_cookie_kwargs  # type: ignore
+            )
+
+
+class Session:
+    def __init__(self, app: Flask) -> None:
+        self.app = app
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app: Flask) -> "None":
+        """This is used to set up session for your app object.
+        :param app: the Flask app object with proper configuration.
+        """
+        app.session_interface = self._get_interface(app)  # type: ignore
+
+    def _get_interface(self, app: Flask) -> SessionInterface:
+        config = app.config.copy()
+        config.setdefault("SESSION_REDIS", Redis())
+        config.setdefault("SESSION_LIFETIME", 2 * 60 * 60)
+        config.setdefault("SESSION_RENEW_COUNT", 5)
+        config.setdefault("SESSION_SIGNER_SALT", "session")
+        config.setdefault("SESSION_KEY_PREFIX", "session:")
+        config.setdefault("SESSION_HEADER_NAME", "authorization")
+
+        session_interface = SessionInterface(
+            config["SESSION_LIFETIME"],
+            config["SESSION_RENEW_COUNT"],
+            config["SESSION_REDIS"],
+            config["SESSION_KEY_PREFIX"],
+            config["SESSION_SIGNER_SALT"],
+            config["SESSION_HEADER_NAME"],
+        )
+
+        return session_interface
+
+
+def logout_user(uid: int) -> None:
+    redis = Redis()
+    for key in redis.keys(app.config["SESSION_KEY_PREFIX"] + "*") + redis.keys(
+        "api_" + app.config["SESSION_KEY_PREFIX"] + "*"
+    ):
+        found = redis.get(key)
+        if found:
+            sess = session_json_serializer.loads(found.decode("utf-8"))
+            if "uid" in sess and sess["uid"] == uid:
+                redis.delete(key)
+
+
+def logout_all() -> None:
+    redis = Redis()
+    for key in redis.keys(app.config["SESSION_KEY_PREFIX"] + "*") + redis.keys(
```

Comment:
similarly as for logout_user(), this will fail with a 500 error

Reply:
https://github.com/freedomofpress/securedrop/pull/6403/commits/64d40726c8f611b137dcfbdb76caf9e6f955bfa4 removes that function as it is never used currently.